front/src/applications/operationalStudies/hooks/useSetupItineraryForTrainUpdate.ts

```diff
@@ -43,23 +92,14 @@ const useSetupItineraryForTrainUpdate = (
   const [postPathProperties] =
     osrdEditoastApi.endpoints.postV2InfraByInfraIdPathProperties.useMutation();
   useEffect(() => {
-    const setupItineraryForTrainUpdate = async () => {
+    const computeItineraryForTrainUpdate = async (
+      trainSchedule: TrainScheduleResult
+    ): Promise<ItineraryForTrainUpdate | null> => {
       if (!infraId) {
-        return;
+        return null;
       }
-      const trainSchedule = await getTrainScheduleById({
-        id: trainIdToEdit,
-      }).unwrap();
       if (!trainSchedule.rolling_stock_name) {
```

Comment:
How do you think about merge the first conditions ?
```typescript
if (!infraId || !trainSchedule.rolling_stock_name) {
        return null;
}
```

Reply:
The first `if (!infraId)` is before the `await getTrainScheduleById()`, and we can only check for `if (!trainSchedule.rolling_stock_name)` after the HTTP request. I prefer to not send an HTTP request if we can avoid it, so that's why I left it that way. (I'd be fine moving both `if`s after the HTTP request if you prefer.)